packages/room/peer/peer.js

```diff
@@ -451,171 +425,210 @@ export const createPeer = ({ api, createStream, event, streams, config }) => {
+            if (
+              systemVideoCodec.mimeType.toLowerCase() ===
+              videoCodec.toLowerCase()
+            ) {
+              preferredCodecs.push(systemVideoCodec)
+              systemVideoCodecs.splice(
+                systemVideoCodecs.indexOf(systemVideoCodec),
+                1
+              )
+            }
+          }
+        }
 
-      const browserName = getBrowserName()
-      const simulcastBrowsers = [SAFARI, CHROME, EDGE, OPERA]
+        // Add all remaining codecs
+        for (const videoCodec of systemVideoCodecs) {
+          preferredCodecs.push(videoCodec)
+        }
 
-      let svc = false
+        /** @type {RTCRtpTransceiverInit} */
+        const transceiverInit = {
+          direction: 'sendonly',
+          streams: [stream.mediaStream],
+        }
 
-      /** @type {RTCRtpCodecCapability[]} */
-      const videoPreferedCodecs = []
-      const videoCodecs = RTCRtpReceiver.getCapabilities('video')?.codecs
+        const svcEnabled =
+          config.media[type].svc &&
+          config.media[type].scalabilityMode &&
+          config.media[type].simulcast &&
+          config.media[type].maxFramerate &&
+          browserName !== FIREFOX
 
-      if (videoCodecs) {
-        if (this._videoCodecPreferences.length > 0) {
-          for (const videoCodecPreference of this._videoCodecPreferences) {
-            for (const videoCodec of videoCodecs) {
-              if (videoCodec.mimeType === videoCodecPreference) {
-                videoPreferedCodecs.push(videoCodec)
+        const simulcastEnabled =
+          config.media[type].simulcast && config.media[type].maxFramerate
 
-                if (
-                  videoCodec.mimeType === 'video/VP9' &&
-                  browserName !== FIREFOX &&
-                  stream.source !== 'screen'
-                ) {
-                  svc = true
-                }
-              }
-            }
+        if (svcEnabled) {
+          /** @type {import('../peer/peer-types.js').RoomPeerType.RTCRtpSVCEncodingParameters} */
+          const scalableEncoding = {
+            maxBitrate: this._highBitrate,
+            scalabilityMode: config.media[type].scalabilityMode,
+            maxFramerate: config.media[type].maxFramerate,
           }
+          transceiverInit.sendEncodings = [scalableEncoding]
+        } else if (simulcastEnabled) {
+          /** @type {RTCRtpEncodingParameters[]} */
+          const simulcastEncoding = [
+            // for firefox order matters... first high resolution, then scaled resolutions...
+            {
+              rid: 'high',
+              maxFramerate: config.media[type].maxFramerate,
+              maxBitrate: this._highBitrate,
+            },
+            {
+              rid: 'mid',
+              // eslint-disable-next-line unicorn/no-zero-fractions
+              scaleResolutionDownBy: 2.0,
+              maxFramerate: config.media[type].maxFramerate,
+              maxBitrate: this._midBitrate,
```

Comment:
When svc is enabled on using simulcast, it must set the scalability mode.

Reply:
Mas @tyohan, when both simulcast and svc are enabled, the sendEncodings config will be like this right?

```js
[{
  rid: 'high',
  maxFramerate: config.media[type].maxFramerate,
  maxBitrate: this._highBitrate,
  scalabilityMode: config.media[type].scalabilityMode
},
{
  rid: 'mid',
  // eslint-disable-next-line unicorn/no-zero-fractions
  scaleResolutionDownBy: 2.0,
  maxFramerate: config.media[type].maxFramerate,
  maxBitrate: this._midBitrate,
  scalabilityMode: config.media[type].scalabilityMode
},
{
  rid: 'low',
  // eslint-disable-next-line unicorn/no-zero-fractions
  scaleResolutionDownBy: 4.0,
  maxFramerate: config.media[type].maxFramerate,
  maxBitrate: this._lowBitrate,
  scalabilityMode: config.media[type].scalabilityMode
}]
```

Reply:
Mas @tyohan. For now, the array send encodings for `addTransceiver()` config will be like these depending on whether simulcast or svc configs are enabled or not. Please let me know if these configs are correct or incorrect.

**1. Svc enabled, simulcast disabled**
```js
[
  {
      "maxBitrate": 1200000,
      "scalabilityMode": "L3T1",
      "maxFramerate": 30
  }
]
```


**2. Svc disabled, simulcast enabled**
```js
[
  {
      "rid": "high",
      "maxFramerate": 30,
      "maxBitrate": 1200000
  },
  {
      "rid": "mid",
      "scaleResolutionDownBy": 2,
      "maxFramerate": 30,
      "maxBitrate": 500000
  },
  {
      "rid": "low",
      "scaleResolutionDownBy": 4,
      "maxFramerate": 30,
      "maxBitrate": 150000
  }
]
```

**3. Svc and simulcast enabled**
```js
[
  {
      "rid": "high",
      "maxFramerate": 30,
      "maxBitrate": 1200000,
      "scalabilityMode": "L3T1"
  },
  {
      "rid": "mid",
      "scaleResolutionDownBy": 2,
      "maxFramerate": 30,
      "maxBitrate": 500000,
      "scalabilityMode": "L3T1"
  },
  {
      "rid": "low",
      "scaleResolutionDownBy": 4,
      "maxFramerate": 30,
      "maxBitrate": 150000,
      "scalabilityMode": "L3T1"
  }
]
```

Reply:
Yes, this is the correct!

Reply:
@faiq-naufal I don't see this changes in peer.js. Inside simulcast encoding I don't see scalabilityMode variable.

Reply:
Mas @tyohan . Please see [on line 593 until 615 at peer.js](https://github.com/inlivedev/inlive-js-sdk/pull/96/files#diff-bbd0bcbca5dd36d5fde9fc44426f7cb5ecc0375d36e9a4bc447dcd12126712f8R593-R615) or the highlighted code below where I included the `scalabilityMode` there.

I made separate checking between simulcast and svc. Because the `scalabilityMode` is svc-related feature, I inject the `scalabilityMode` option when the svc config is enabled. I also checked whether the simulcast config is also enabled. If enabled, then the `scalabilityMode` will be appended to the simulcast config.

```js
if (simulcastEnabled) {
  /** @type {RTCRtpEncodingParameters[]} */
  let simulcastEncoding = [
    // for firefox order matters... first high resolution, then scaled resolutions...
    {
      rid: 'high',
      maxFramerate: config.media[type].maxFramerate,
      maxBitrate: this._highBitrate,
    },
    {
      rid: 'mid',
      // eslint-disable-next-line unicorn/no-zero-fractions
      scaleResolutionDownBy: 2.0,
      maxFramerate: config.media[type].maxFramerate,
      maxBitrate: this._midBitrate,
    },
    {
      rid: 'low',
      // eslint-disable-next-line unicorn/no-zero-fractions
      scaleResolutionDownBy: 4.0,
      maxFramerate: config.media[type].maxFramerate,
      maxBitrate: this._lowBitrate,
    },
  ]

  transceiverInit.sendEncodings = simulcastEncoding
}

if (svcEnabled) {
  if (
    simulcastEnabled &&
    Array.isArray(transceiverInit.sendEncodings)
  ) {
    const sendEncodings = transceiverInit.sendEncodings.map(
      (encoding) => {
        return {
          ...encoding,
          scalabilityMode: config.media[type].scalabilityMode,
        }
      }
    )
    transceiverInit.sendEncodings = sendEncodings
  } else {
    const sendEncodings = {
      maxBitrate: this._highBitrate,
      scalabilityMode: config.media[type].scalabilityMode,
      maxFramerate: config.media[type].maxFramerate,
    }
    transceiverInit.sendEncodings = [sendEncodings]
  }
}

const videoTransceiver = this._peerConnection.addTransceiver(
  videoTrack,
  transceiverInit
)
```